#Import Libraries

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix

#Load Dataset

In [ ]:
df = pd.read_csv('cleaned_titanic.csv')

df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",1,-0.565736,1,0,A/5 21171,-0.502445,2
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",0,0.663861,1,0,PC 17599,0.786845,0
2,3,1,3,"Heikkinen, Miss. Laina",0,-0.258337,0,0,STON/O2. 3101282,-0.488854,2
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",0,0.433312,1,0,113803,0.420730,2
4,5,0,3,"Allen, Mr. William Henry",1,0.433312,0,0,373450,-0.486337,2


#Explore Dataset

In [ ]:
df.info()

df.describe()

df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    int64  
 5   Age          891 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Embarked     891 non-null    int64  
dtypes: float64(2), int64(7), object(2)
memory usage: 76.7+ KB


,0
PassengerId,0
Survived,0
Pclass,0
Name,0
Sex,0
Age,0
SibSp,0
Parch,0
Ticket,0
Fare,0


The Titanic dataset was explored to understand its structure, data types, and missing values before building the machine learning model.

#Select Features and Target

In [ ]:
X = df[['Pclass', 'Sex', 'Age', 'Fare']]

y = df['Survived']

In [ ]:
print(X.head())
print(X.isnull().sum())

   Pclass  Sex       Age      Fare
0       3    1 -0.565736 -0.502445
1       1    0  0.663861  0.786845
2       3    0 -0.258337 -0.488854
3       1    0  0.433312  0.420730
4       3    1  0.433312 -0.486337
Pclass    0
Sex       0
Age       0
Fare      0
dtype: int64


#Split the Dataset

In [ ]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42
)

#Train Logistic Regression

In [ ]:
from sklearn.linear_model import LogisticRegression

model = LogisticRegression()

model.fit(X_train, y_train)

LogisticRegression()

#Make Predictions

In [ ]:
y_pred = model.predict(X_test)

print(y_pred[:10])

[0 0 0 1 1 1 1 0 1 1]


#Evaluate the Model

In [ ]:
from sklearn.metrics import accuracy_score

accuracy = accuracy_score(y_test, y_pred)

print("Accuracy:", accuracy)

Accuracy: 0.7988826815642458


#confusion matrix

In [ ]:
from sklearn.metrics import confusion_matrix

cm = confusion_matrix(y_test, y_pred)

print(cm)

[[90 15]
 [21 53]]


This means

90 passengers were correctly predicted as not surviving.
53 passengers were correctly predicted as surviving.
15 passengers were incorrectly predicted as surviving.
21 passengers were incorrectly predicted as not surviving.

#classification report:

In [ ]:
from sklearn.metrics import classification_report

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.81      0.86      0.83       105
           1       0.78      0.72      0.75        74

    accuracy                           0.80       179
   macro avg       0.80      0.79      0.79       179
weighted avg       0.80      0.80      0.80       179



Model Performance
Accuracy Score: 79.89%
Precision: 81% (Class 0), 78% (Class 1)
Recall: 86% (Class 0), 72% (Class 1)
F1 Score: 83% (Class 0), 75% (Class 1)

#CONCLUSION
The Logistic Regression model achieved an accuracy score of 79.89% on the Titanic dataset. The model performed well in predicting passenger survival based on Pclass, Sex, Age, and Fare. The confusion matrix and classification report indicate that the model correctly classified most passengers and can be considered a reliable baseline classification model.

#Key Findings
Female passengers had a higher chance of survival.
Passenger class significantly influenced survival rates.
Logistic Regression provided good classification performance with nearly 80% accuracy.